### DISCUSSION TITLE
error with feature case: ** (MatchError) no match of right hand side value: {:already, :allowed}

### DISCUSSION THREAD

**Discussion by hoangbits** [keathley/wallaby#653] [Q&A] [answered]
I want to put `use Wallaby.Feature` inside a FeatureCase (similar role to ConnCase),
But then when I use it in a test module with: `use AppWeb.FeatureCase, async: true`
at: https://github.com/hoangbits/wallaby-bug-report1/blob/c5bbfd2cf5f0f683a934685ffb06b95893aae329/test/app_web/integration/index_test.exs#L2
I got the error.

Detail stacktrace: 
```
➜ mix test test/app_web/integration                    

  1) feature renders index (AppWeb.Integration.IndexTest)
     test/app_web/integration/index_test.exs:17
     ** (MatchError) no match of right hand side value: {:already, :allowed}
     stacktrace:
       (wallaby 0.29.1) lib/wallaby/feature.ex:174: Wallaby.Feature.Utils.checkout_ecto_repos/2
       (elixir 1.12.3) lib/enum.ex:1582: Enum."-map/2-lists^map/1-0-"/2
       (wallaby 0.29.1) lib/wallaby/feature.ex:164: Wallaby.Feature.Utils.maybe_checkout_repos/1
       test/app_web/integration/index_test.exs:2: AppWeb.Integration.IndexTest.__ex_unit_setup_1/1
       test/app_web/integration/index_test.exs:1: AppWeb.Integration.IndexTest.__ex_unit__/2
```
Github repo that can be used to reproduce:  
- https://github.com/hoangbits/wallaby-bug-report1
FeatureCase file:
- https://github.com/hoangbits/wallaby-bug-report1/blob/main/test/support/feature_case.ex

I really appreciate any help.

**Comment by mhanberg** [comment, marked as answer]
You don't need any of the stuff in the setup block of your ExUnit case template, as wallaby takes care of that for you when you do `use Wallaby.Feature`

**Reply by hoangbits** [reply]
So I remove `use Wallaby.Feature` from CaseTemplate and bring it  `use Wallaby.Feature` to test module.
inside test module: 
```
defmodule AppWeb.Integration.IndexTest do
  use AppWeb.FeatureCase, async: true
  use Wallaby.Feature
```
But the same error was still raised as long as I keep ` async: true` like above.

**Reply by mhanberg** [reply]
I am saying to remove the setup block, not the call to `use Wallaby.Feature`. 

When you do `use Wallaby.Feature`, it will automatically checkout the ecto repo as well as start a session. 

You are getting your original error because you are checking out the repo a second time.

**Reply by hoangbits** [reply]
```
You are getting your original error because you are checking out the repo a second time.
```
 it's clear to me now.
Thank you very much.
